### ISSUE TITLE
[Bug]: primary key duplicate check miscalculation

### ISSUE THREAD

**Issue by zhangxu19830126** [matrixorigin/matrixone#10135]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

#10027 cause primary key duplicate check miscalculation in RC+ pessimistic mode. And revert this pr，all is ok

**Comment by XuPeng-SH** [comment]
assign to @aunjgr who is the owner of #10027

**Comment by aunjgr** [comment]
Is it fixed by #10145?
